pynetsnmp/twistedsnmp.py

```diff
@@ -309,7 +310,21 @@ def _getCmdLineArgs(self):
         return cmdLineArgs
 
     def open(self):
-        self.session = netsnmp.Session(cmdLineArgs=self._getCmdLineArgs())
+        if self.session is not None:
+            self.session.close()
+            self.session = None
+
+        self.session = netsnmp.Session(
+            version=netsnmp.SNMP_VERSION_MAP.get(
+                self.snmpVersion, 
+                netsnmp.SNMP_VERSION_2c),
+            timeout=int(self.timeout),
+            retries=int(self.tries),
+            peername= '%s:%d' % (self.ip, self.port),
+            community=self.community,
+            community_len=len(self.community),
+            cmdLineArgs=self._getCmdLineArgs())
```

Comment:
The way `initialize_session` (in netsnmp.py) is written, if there are any arguments given in `cmdLineArgs`, they are overridden by the other keyword arguments.  Conceptually, command line arguments should override default values or values provided by a configuration file.  On its face, this change would appear to upset that expectation.

Can you show me that this is not the case?

Reply:
Looks like none of the places where cmdLineArgs is passed will be effected by this change.